include/swift/Runtime/RuntimeFunctions.def

```diff
@@ -1225,12 +1225,26 @@ FUNCTION(EndAccess, swift_endAccess, C_CC, AlwaysAvailable,
          ARGS(getFixedBufferTy()->getPointerTo()),
          ATTRS(NoUnwind))
 
-FUNCTION(GetOrigOfReplaceable, swift_getOrigOfReplaceable, C_CC, AlwaysAvailable,
+FUNCTION(GetOrigOfReplaceable, swift_getOrigOfReplaceable, C_CC,
+         GetReplacementAvailability,
          RETURNS(FunctionPtrTy),
          ARGS(FunctionPtrTy->getPointerTo()),
          ATTRS(NoUnwind))
 
-FUNCTION(GetReplacement, swift_getFunctionReplacement, C_CC, AlwaysAvailable,
+FUNCTION(GetReplacement, swift_getFunctionReplacement, C_CC,
+         GetReplacementAvailability,
+         RETURNS(FunctionPtrTy),
+         ARGS(FunctionPtrTy->getPointerTo(), FunctionPtrTy),
+         ATTRS(NoUnwind))
+
+FUNCTION(GetOrigOfReplaceable50, swift_getOrigOfReplaceable50, C_CC,
+         AlwaysAvailable,
+         RETURNS(FunctionPtrTy),
+         ARGS(FunctionPtrTy->getPointerTo()),
+         ATTRS(NoUnwind))
+
+FUNCTION(GetReplacement50, swift_getFunctionReplacement50, C_CC,
```

Comment:
I had been thinking about this issue of backporting new entry points with the compatibility library, and I wonder whether there might be a more systematic way of going about it. I was thinking that maybe we could turn the return type of the availability function here from a `bool` to a tristate, `ConditionallyAvailable` | `AvailableByCompatibilityLibrary` | `AlwaysAvailable`. We could factor the logic for choosing the compatibility library or OS version of the entry point into `getRuntimeFn` then, instead of repeating this conditional logic for every client of the entry point.